Framework/API/src/AlgorithmProperties.cpp

```diff
@@ -47,6 +47,12 @@ void update(std::string const &property, boost::optional<double> const &value, I
     update(property, value.get(), properties);
 }
 
+void update(std::string const &property, MatrixWorkspace_sptr const &workspace, IAlgorithmRuntimeProps &properties) {
+  if (workspace) {
```

Comment:
Do you need this `if (workspace)`? It was perhaps present in the preceding function because the `value` parameter was a `boost::optional`